plugins/modules/system/sysupgrade.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2020, Andrew Klaus <andrewklaus@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: sysupgrade
+
+short_description: Manage OpenBSD system upgrades
+
+version_added: "2.10"
+
+description:
+    - "Manage OpenBSD system patches using sysupgrade"
+
+options:
+    snapshot:
+        description:
+            - Apply latest snapshot. Otherwise release will be applied.
+        default: false
+        required: false
+        type: bool
+    force:
+        description:
+            - Force upgrade (for snapshots only).
+        required: false
+        type: bool
+    keep_files:
+        description:
+            - Keep the files in /home/_sysupgrade. By default they will be deleted after the upgrade.
+        required: false
+        type: bool
+    fetch_only:
+        description:
+            - Fetch and verify files and create /bsd.upgrade but do not reboot.
+        default: true
+        required: false
+        type: bool
+
+author:
+    - Andrew Klaus (@precurse)
+'''
+
+EXAMPLES = '''
+- name: Upgrade to latest release
+  sysupgrade:
+  become: true
+  register: sysupgrade
+
+- name: Upgrade to latest snapshot
+  sysupgrade:
+    snapshot: true
+  register: sysupgrade
+
+- name: Have Ansible reboot if required
+  reboot:
+  when: sysupgrade.changed
+
+# Note: Ansible will error when running this way due to how
+#   the reboot is forcefully handled by sysupgrade:
+
+- name: Have sysupgrade automatically reboot
+  sysupgrade:
+    fetch_only: false
+  become: true
+  register: sysupgrade
+  ignore_errors: true
+'''
+
+RETURN = r'''
+rc:
+  description: The command return code (0 means success)
+  returned: always
+  type: int
+stdout:
+  description: sysupgrade standard output
+  returned: always
+  type: str
+stderr:
+  description: sysupgrade standard error
+  returned: always
+  type: str
+  sample: "sysupgrade: need root privileges"
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def run_module():
+    # define available arguments/parameters a user can pass to the module
+    module_args = dict(
+        snapshot=dict(type='bool', default=False),
+        fetch_only=dict(type='bool', default=True),
+        force=dict(type='bool', default=False),
+        keep_files=dict(type='bool', default=False),
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=False,
+    )
+
+    result = sysupgrade_run(module)
+
+    module.exit_json(**result)
+
+
+def sysupgrade_run(module):
+    cmd = ['/usr/sbin/sysupgrade']
```

Comment:
```suggestion
    sysupgrade_bin = module.get_bin_path('/usr/sbin/sysupgrade', required=True)
    cmd = [sysupgrade_bin]
```
This way it will check existence of sysupgrade binary.